### ISSUE TITLE
Phase Hook behavior not fully documented

### ISSUE THREAD

**Issue by jimu** [boardgameio/boardgame.io#808]
[Phases Documentation](https://boardgame.io/documentation/#/phases) should include:

1) the order that hooks are executed
2) the conditions under which they are executed

See 
  * issue #569 (2020-03-20)
  * [gitter message](https://gitter.im/boardgame-io/General?at=5f6bd4ffefefe620f245430a)

**Comment by delucis** [comment]
Agreed that these would be very helpful to add — I’m not even 100% clear on order myself.

**Comment by delucis** [comment]
These tests might be a good place to start:

https://github.com/boardgameio/boardgame.io/blob/7254749886d7b6c1415a5dc2abeccf03260833cf/src/core/flow.test.ts#L1521-L1731

**Comment by delucis** [comment]
![bgio-hooks-order](https://user-images.githubusercontent.com/357379/137633133-09560909-aeb5-4aa0-b468-ba3cccbb7087.png)
